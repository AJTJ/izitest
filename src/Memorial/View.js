import React from "react";
import parse from "date-fns/parse";
import format from "date-fns/format";

//styling
import { Col } from "reactstrap";

import { MemorialText, memorialContainer } from "./styles";

const Memorial = ({ name, creationDate }) => {
  return (
    <Col className={`${memorialContainer}`} xs={11} sm={5} md={5} lg={3}>
      {name ? (
        <MemorialText fontSize={"20px"} paddingbottom={"30px"}>
          {name.last}, {name.first}
        </MemorialText>
      ) : (
        <MemorialText fontSize={"20px"} paddingbottom={"30px"}>
          Unnamed
        </MemorialText>
      )}
      <MemorialText>{format(parse(creationDate), "MM/DD/YYYY")}</MemorialText>
    </Col>
  );
};

export default Memorial;
