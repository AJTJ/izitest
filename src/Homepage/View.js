import React from "react";

import { connect } from "react-redux";
import { actions } from "../redux";

import { Container, Row } from "reactstrap";

import Memorial from "../Memorial/View";
import { Button } from "../components/buttons";
import { Hero, Title } from "./styles";

const Homepage = props => {
  return (
    <React.Fragment>
      <Hero fluid>
        <Row className={`d-flex justify-content-center`}>
          <Title>Memorials</Title>
        </Row>
        <Row className="d-flex justify-content-center">
          <Button
            onClick={() =>
              props.getMemorials({
                url: "https://dev.requiemapp.com/public/memorial/random"
              })
            }
          >
            Get memorials
          </Button>
          <Button onClick={() => props.orderMemorials()}>
            Order by last name
          </Button>
        </Row>
      </Hero>
      <Container>
        <Row className="d-flex justify-content-center">
          {props.memorials.map((memorial, key) => {
            const { name, creationDate } = memorial;
            return (
              <React.Fragment key={key}>
                <Memorial
                  name={name || undefined}
                  creationDate={creationDate}
                />
              </React.Fragment>
            );
          })}
        </Row>
      </Container>
    </React.Fragment>
  );
};

const mapStateToProps = state => {
  return { memorials: state.memorialsReducer.memorials };
};

const mapDispatchToProps = {
  ...actions
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Homepage);
