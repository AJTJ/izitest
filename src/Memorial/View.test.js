import React from "react";
import { default as Component } from "./View";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const comp = renderer
    .create(
      <Component
        name={{ last: "testLast", first: "testFirst" }}
        creationDate={873873873}
      />
    )
    .toJSON();
  expect(comp).toMatchSnapshot();
});
