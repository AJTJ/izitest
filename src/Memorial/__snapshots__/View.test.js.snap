// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-muw0v4 col-11 col-sm-5 col-md-5 col-lg-3"
>
  <div
    className="css-12qzm3b row"
    fontSize="20px"
    paddingbottom="30px"
  >
    testLast
    , 
    testFirst
  </div>
  <div
    className="css-odlume row"
  >
    01/10/1970
  </div>
</div>
`;
